libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -192,11 +223,118 @@ bool AP_OABendyRuler::update(const Location& current_loc, const Location& destin
     destination_new.offset_bearing(chosen_bearing, distance_to_dest);
 
     // log results
-    AP::logger().Write_OABendyRuler(true, chosen_bearing, false, best_margin, destination, destination_new);
+    AP::logger().Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_HORIZONTAL, true, chosen_bearing, 0.0f, false, best_margin, destination, destination_new);
+
+    return true;
+}
+
+// Search for path in the Vertical directions
+bool AP_OABendyRuler::search_vertical_path(const Location& current_loc, const Location& destination,Location &destination_new, const float &lookahead_step1_dist, const float &lookahead_step2_dist, const float &bearing_to_dest, const float &distance_to_dest) 
+{
+    // check OA_BEARING_INC_VERTICAL definition allows checking in all directions
+    static_assert(360 % OA_BENDYRULER_BEARING_INC_VERTICAL == 0, "check 360 is a multiple of OA_BEARING_INC_VERTICAL");
+    float best_vertical_bearing = 0.0f;
+    bool  have_best_vertical_bearing = false;
+    float best_margin = -FLT_MAX;
+    float best_margin_pitch = best_vertical_bearing;
+    const uint8_t limit = 180 / OA_BENDYRULER_BEARING_INC_VERTICAL;
```

Comment:
let's rename "limit" to "angular_limit" or something like that.  In any case, having this temp variable is better than what the existing horizontal equivalent does.